### DISCUSSION TITLE
How to enter the docker container ? I have tried many times ,but failed

### DISCUSSION THREAD

**Discussion by marchawake** [bluenviron/mediamtx#489] [Questions]
## Which version are you using?

latest

## Which operating system are you using?

OS

- [ ] Linux

Architecture

- [ ] arm64

root@storageperfect-desktop:~# docker exec -it rtsp_server bash
OCI runtime exec failed: exec failed: container_linux.go:349: starting container process caused "exec: \"bash\": executable file not found in $PATH": unknown
root@storageperfect-desktop:~# docker exec -it rtsp_server sh
OCI runtime exec failed: exec failed: container_linux.go:349: starting container process caused "exec: \"sh\": executable file not found in $PATH": unknown
root@storageperfect-desktop:~# docker exec -it rtsp_server /bin/sh
OCI runtime exec failed: exec failed: container_linux.go:349: starting container process caused "exec: \"/bin/sh\": stat /bin/sh: no such file or directory": unknown
root@storageperfect-desktop:~# docker exec -it 5b2fd8491969 /bin/bash

**Comment by dorinclisu** [comment]
It's not possible to enter the container because the server image is built from scratch instead of a OS base image containing basic binaries like bash and stuff.

**Comment by aler9** [comment]
The default image doesn't include a shell in order to make it as small as possible, but you are free to build a new image that includes a shell:

```
docker build -t myimage << EOF
FROM aler9/rtsp-simple-server AS server
FROM alpine:3.13
COPY --from=server /rtsp-simple-server /rtsp-simple-server
ENTRYPOINT [ "/rtsp-simple-server" ]
```

[a comment by an automated account is not shown]
